File: #0404.sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {

        return !root ? 0 :
            (root->left && !root->left->left
            && !root->left->right ? root->left->val : 0) + 
                sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right);
    }
};

File: #0404.sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
    if (root == null) {
        return 0;
    }

    let result = 0;
    function sumOfLeftLeaves(root , isLeft) {
        if (isLeft && !root.left && !root.right) {
            result += root.val;
        }

        if (root.left) {
            sumOfLeftLeaves(root.left , true);
        }

        if (root.right) {
            sumOfLeftLeaves(root.right , false);
        }
    }

    sumOfLeftLeaves(root , false);
    return result;
};
